### ISSUE TITLE
Folder patterns is not recognized properly

### ISSUE THREAD

**Issue by mpospelov** [toptal/codeowners-checker#72]
### Steps to reproduce:

```
#!/bin/bash
mkdir .github/
printf "lib/ @jonatas\nspec/ @other" > .github/CODEOWNERS
printf "@jonatas\n@other" > .github/OWNERS
touch lib/foobar.rb
touch spec/foobar.rb
git add lib spec
git commit -m "test"
bin/codeowners-checker check .
```

---

### Actual behavior: 
```
# NOTE: Gem recognize new file as the missing owner but they should be identified by the pattern
File added: "lib/foobar.rb". Add owner to the CODEOWNERS file?
(y) yes
(i) ignore
(q) quit and save
 [y, i, q] y
Owners:
1 - @jonatas
2 - @other
Choose owner, add new one or leave empty to use "".
New owner:  1
Possible groups to which the pattern belongs:
# NOTE: Gem recognizes the possible group for that file but for spec/foobar.rb it's missing
1 - lib/ @jonatas
Choose group:  1
File added: "spec/foobar.rb". Add owner to the CODEOWNERS file?
(y) yes
(i) ignore
(q) quit and save
 [y, i, q] y
Owners:
1 - @jonatas
2 - @other
Choose owner, add new one or leave empty to use "".
New owner:  2
Add to the end of the CODEOWNERS file?
Commit changes? n
```

---

### Expected: 
as we have matching owners for these new files via patterns I think codeowners shouldn't complain about that so the next output seems what we need to have in this case
```
✅ File is consistent
```

**Comment by paneq** [comment]
In fact, even directory name without `/` is supported by Github so the scenario described below should also be supported.

```
#!/bin/bash
mkdir .github/
printf "lib @jonatas\nspec @other" > .github/CODEOWNERS
printf "@jonatas\n@other" > .github/OWNERS
touch lib/foobar.rb
touch spec/foobar.rb
git add lib spec
git commit -m "test"
bin/codeowners-checker check .
```

**Comment by paneq** [comment]
Basically we should first check if the pattern is a directory name and in such case apply different behavior. This will require quite some changes to the library.

https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

**Comment by paneq** [comment]
This is the best way that I found so far for interpreting which rules evaluate to which files:

```
require 'find'
require 'set'

rules = File.
  read(".github/CODEOWNERS").
  lines.
  reject{|l| l.strip.empty? || l.start_with?("#") }.
  map{|l| l.split.first }.
  map(&:strip)

known_files = rules.flat_map do |rule|
  rule = rule[1..-1] if rule.start_with?("/")
  Dir.glob(rule).flat_map do |f|
    if File.directory?(f)
      Find.find(f).to_a
    else
      f
    end
  end
end.to_set;nil
```

**Comment by dennissivia** [comment]
The problem has been fixed in https://github.com/toptal/codeowners-checker/pull/76, by using `pathspec` for full `gitignore`-syntax support.
